github/codeowners.py

```diff
@@ -27,54 +27,56 @@
 ci.log.configure_default_logging()
 
 
-class CodeownersEnumerator:
-    '''
-    Parses GitHub CODEOWNSERS files [0] from the documented default locations for a given
-    (git) repository work tree into a stream of codeowners entries.
+def enumerate_remote_repo(
+    repo: github3.repos.repo.Repository,
+    paths: typing.Sequence[str] = ('CODEOWNERS', '.github/CODEOWNERS', 'docs/CODEOWNERS'),
+) -> typing.Generator[str, None, None]:
+    for path in paths:
+        try:
+            yield from filter_codeowners_entries(
+                repo.file_contents(path=path).decoded.decode('utf-8').split('\n')
+            )
+        except NotFoundError:
+            pass # ignore absent files
 
-    [0] https://help.github.com/articles/about-codeowners/
+
+def enumerate_single_file(
+    file_path: str,
+):
+    file_path = existing_file(file_path)
+    with open(file_path) as f:
+        yield from filter_codeowners_entries(f.readlines())
+
+
+def enumerate_local_repo(
+    repo_dir: str,
+    paths: typing.Sequence[str] = ('CODEOWNERS', '.github/CODEOWNERS', 'docs/CODEOWNERS'),
+):
+    repo_dir = existing_dir(Path(repo_dir))
+    if not repo_dir.joinpath('.git').is_dir():
+        raise ValueError(f'not a git root directory: {repo_dir}')
+
+    for path in paths:
+        codeowners_file = repo_dir.joinpath(path)
+        if codeowners_file.is_file():
+            with open(codeowners_file) as f:
+                yield from filter_codeowners_entries(f.readlines())
+
+
+def filter_codeowners_entries(lines):
```

Comment:
generally speaking, I would prefer typehints to be concise. It will be very rarely be the case that we do actually require a `sequence` (I think most notably, sequences define a fixed order of elements. by stating we need a sequence, we would e.g. "forbid" sets to be passed)